src/core/geometry/qgsquadix.cpp

```diff
@@ -0,0 +1,291 @@
+/***************************************************************************
+                         qgsquadix.cpp
+                         -------------------
+    begin                : November 2018
+    copyright            : (C) 2018 by Loïc Bartoletti
+    email                : loic dot bartoletti at oslandia dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsquadix.h"
+#include "qgsgeometryutils.h"
+
+QgsQuadix::QgsQuadix()
+{
+
+}
+
+QgsQuadix::QgsQuadix( const QgsPoint &p1, const QgsPoint &p2, const QgsPoint &p3, const QgsPoint &p4 ):
```

Comment:
QGIS code convention is:

```
QgsQuadix::QgsQuadix( const QgsPoint &p1, const QgsPoint &p2, const QgsPoint &p3, const QgsPoint &p4 )
    : mPoint1( p1 )
    , mPoint2( p2 )
    , etc
```

This helps keep diffs cleaner if the initializers are changed in future.